### DISCUSSION TITLE
v1.41.0_64-dev

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#1277] [Announcements]
# v1.41.0_64-dev

## Highlight - Public sharing 🎉

Yes, one more time, I am very proud to present the new awaiting feature of Immich - **Public Sharing**. It is quite similar to Google Photos public shared link, however, with even more features.

![image](https://user-images.githubusercontent.com/27055614/211461647-9faa3909-7c60-4616-b58f-f2cbf83f2c28.png)

Each album can have multiple links; each link can be individually configured based on the permission to upload or the expiration time. You can manage all the links in the shared link page, not just the shared link for the group of assets.

![image](https://user-images.githubusercontent.com/27055614/211461924-dbae7d22-75d7-4918-b682-a18d862e9e63.png)

As a public user, he/she can
 
* View individual assets.
* Download a group of assets.
* Download the whole shared album.
* Upload asset (if allowed) - the asset will belong to the user that shares the link.

We've made this specific page to be mobile friendly so it can be viewed correctly on every screen on the mobile browser, the download and upload feature also works on the mobile browser.

We plan to add the functionality to share individual or group assets in the next release. 

Please help us test out this new feature and let me know your thoughts or report bugs that you found.

Besides the abovementioned feature, we have deployed a TON, yes! a TON more of bug fixes and features. Please read the detailed changed log below.

As always, please consider supporting the project.

🎉 Cheer! 🎉

## Support

<p align="center">
<img src="https://media.giphy.com/media/LStqgGESXW8XnuCv5y/giphy.gif" width="250" title="Loading ~4000 images/videos"> 
</p>

If you find the project helpful and it helps you in some ways, you can support the project [one time](https://github.com/sponsors/alextran1502?frequency=one-time&sponsor=alextran1502) or [monthly](https://github.com/sponsors/alextran1502) from GitHub Sponsor

It is a great way to let me know that you want me to continue developing and working on this project for years to come.

## What's Changed
* fix(nginx): define stopsignal by @otbutz in https://github.com/immich-app/immich/pull/1207
* feat(server): mobile oauth with custom scheme redirect uri by @jrasm91 in https://github.com/immich-app/immich/pull/1204
* fix(docker-build): start main process with exec by @bt90 in https://github.com/immich-app/immich/pull/1210
* chore(web): show corresponding ffmpeg argument by @jrasm91 in https://github.com/immich-app/immich/pull/1217
* fix(web) Fix incorrect album type (shared/non-shared) in album selection modal by @alextran1502 in https://github.com/immich-app/immich/pull/1219
* feat(web) Add drag n drop upload functionality by @klejejs in https://github.com/immich-app/immich/pull/1216
* feat(nginx): Enable upstream keepalive by @otbutz in https://github.com/immich-app/immich/pull/1206
* feat(nginx): Compress more types by @bt90 in https://github.com/immich-app/immich/pull/1211
* Docs: Github -> GitHub by @Emalton in https://github.com/immich-app/immich/pull/1221
* fix nginx conf file by @vedhavyas in https://github.com/immich-app/immich/pull/1229
* refactor(server): imports and repository tokens by @jrasm91 in https://github.com/immich-app/immich/pull/1220
* feat(deployment): Consistent container names by @bt90 in https://github.com/immich-app/immich/pull/1232
* fix(web): hide deleted users by @jrasm91 in https://github.com/immich-app/immich/pull/1237
* feat(web,server): api keys by @jrasm91 in https://github.com/immich-app/immich/pull/1244
* fix(nginx): Fix connection header by @bt90 in https://github.com/immich-app/immich/pull/1252
* fix(server): log user delete errors by @jrasm91 in https://github.com/immich-app/immich/pull/1255
* feat(web/server) public album sharing by @alextran1502 in https://github.com/immich-app/immich/pull/1266
* feat(web): Add search functionality to add photo to album modal by @klejejs in https://github.com/immich-app/immich/pull/1273
* chore(web) Add loading indicators to jobs panel by @klejejs in https://github.com/immich-app/immich/pull/1272
* chore(web) Add automatic server stats refetching by @klejejs in https://github.com/immich-app/immich/pull/1271
* feat(web,server): disable password login by @jrasm91 in https://github.com/immich-app/immich/pull/1223
* fix(mobile): omit null values from json requests by @jrasm91 in https://github.com/immich-app/immich/pull/1212

## New Contributors
* @otbutz made their first contribution in https://github.com/immich-app/immich/pull/1207
* @klejejs made their first contribution in https://github.com/immich-app/immich/pull/1216
* @Emalton made their first contribution in https://github.com/immich-app/immich/pull/1221
* @vedhavyas made their first contribution in https://github.com/immich-app/immich/pull/1229

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.40.1_63-dev...v1.41.0_64-dev

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.41.0_64-dev'>v1.41.0_64-dev</a>.</em>

**Comment by AvMavs** [comment]
This is so well thought out!

**Comment by 0rn0lf** [comment]
Awesome release. Thanks for your hard work. 
The public sharing function is pretty amazing. Would it be possible to add a function to restrict the users from downloading assets/albums?

**Reply by bo0tzz** [reply]
Restricting downloads doesn't make sense, since the users are already receiving the file to view it and could also just as easily take a screenshot or such.

**Reply by 0rn0lf** [reply]
While this is particular true, it’s much more inconvenient to do this with a hole album. Maybe you just want to show your creations but don’t want to share the original file.

**Reply by jpmiller25** [reply]
I'll add, this is a feature commonly used by photographers with their shared albums, many times they will share the low res or thumbnails as proofs but restrict downloads of the full res behind a paywall. Though this app is not targeting that user base and the photogs I know use specialty galleries designed for that use case.

**Reply by alextran1502** [reply]
I will consider this, thank you for the feedback!

**Reply by jrasm91** [reply]
That makes more sense to me. So something like `Download Originals` true allows access to the original file, otherwise they'll just see/get the resized version.

**Comment by sohail579** [comment]
This feature is a game changer, Thankyou @alextran1502  and the Immich team.  I have sucessfully shared many links today :)  however can only do so through the web interface not the app is this correct?

**Reply by alextran1502** [reply]
That is correct

**Reply by sohail579** [reply]
Thanks for clarifying thought there was something wrong with the app

**Comment by lxrst** [comment]
Idea: add a cancel button to the batch download toast

**Reply by bo0tzz** [reply]
I like the suggestion, can you create an issue for that?

**Reply by lxrst** [reply]
@bo0tzz Thanks! #1291

**Comment by bt90** [comment]
With the latest version I have completed my crusade against unresponsive container restarts. Stopping containers should now no longer take 10s.

**Reply by jrasm91** [reply]
Thank you for your contributions!
